org.lflang/src/org/lflang/generator/CppGenerator.kt

```diff
@@ -0,0 +1,263 @@
+ * are permitted provided that the following conditions are met:
+
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ ***************/
+
+package org.lflang.generator
+
+import org.eclipse.emf.ecore.resource.Resource
+import org.eclipse.xtext.generator.IFileSystemAccess2
+import org.eclipse.xtext.generator.IGeneratorContext
+import org.lflang.*
+import org.lflang.Target
+import org.lflang.lf.Action
+import org.lflang.lf.Reactor
+import org.lflang.lf.VarRef
+import java.nio.file.Path
+
+/* *******************************************************************************************
+ * The following definitions are shortcuts to access static members of FileConfig and ASTUtils
+ *
+ * TODO these should likely be moved to a common place in the future
+ */
+
+val Resource.name: String get() = FileConfig.getName(this)
+
+fun Path.toUnixString(): String = FileConfig.toUnixString(this)
+
+val Reactor.isGeneric get() = ASTUtils.isGeneric(this.toDefinition())
+
+/* ********************************************************************************************/
+
+/** Prepend each line of the rhs multiline string with the lhs prefix.
+ *
+ * This is a neat little trick that allows for convenient insertion of multiline strings in string templates
+ * while correctly managing the indentation. Consider this multiline string:
+ * ```
+ * val foo = """
+ *    void foo() {
+ *        // do something useful
+ *    }""".trimIndent()
+ * ```
+ *
+ * It could be inserted into a higher level multiline string like this:
+ *
+ * ```
+ * val bar = """
+ *     |class Bar {
+ * ${" |    "..foo}
+ *     |};""".trimMargin()
+ * ```
+ *
+ * This will produce the expected output:
+ * ```
+ * class Bar {
+ *     void foo() {
+ *         // do something useful
+ *     }
+ * };
+ * ```
+ */
+operator fun String.rangeTo(str: String) = str.prependIndent(this)
```

Comment:
Should this be a call to `replaceIndent` instead? Otherwise the common indent of the `str` parameter is preserved. I'm not sure, this matters in the use cases you have below, but this could make a difference in the future

Reply:
Probably, yes. I didn't know about `replaceIndent`